<?xml version="1.0"?>

<launch>  
    <arg name="fake" default="false" />
    <arg name="rviz" default="true" />
    <arg name="show_pose" default="false" />
    <arg name="show_traj" default="false" />
    <arg name="hmi" default="true" />  <!--FIXME-->

    <!-- robot_ip: IP-address of the robot's socket-messaging server -->
    <!-- <arg name="robot_ip" doc="IP of the controller" /> -->

    <!-- robot driver -->
    <include unless="$(arg fake)" file="$(find oil_driver)/launch/oil_driver.launch"> 
      <arg name="use_hand" value="false" />
      <arg name="show_pose_info" value="$(arg show_pose)" />
      <arg name="show_traj_info" value="$(arg show_traj)" />
    </include>

    <!-- robot model -->
    <include unless="$(arg fake)" file="$(find oil_moveit_config)/launch/demo.launch">
      <arg name="use_rviz" value="$(arg rviz)" />
    </include>

    <!-- robot model fake -->
    <include if="$(arg fake)" file="$(find oil_moveit_config)/launch/fake.launch">
      <arg name="use_rviz" value="$(arg rviz)" />
      <arg name="use_publisher" value="true" />
      <arg name="use_gui" value="false" />
    </include>

    <!-- robot hmi -->
    <!-- <include if="$(arg hmi)" file="$(find oil_hmi)/launch/oil_hmi.launch" /> -->

    <!-- move server -->
    <node name="oil_move_server" pkg="oil_application" type="oil_move_server.py" respawn="true" output="screen" />

</launch>
